src/utils/convert3Dto4D.m

```diff
@@ -0,0 +1,200 @@
+        newline;
+
+      else
+
+        fprintf('\n\nCONVERTING SEQUENCE: %s \n\n', char(optSource.sequenceList(iSeq)));
+
+        % Set whether to remove dummies or not
+
+        nbDummies = 0;
+
+        if contains(optSource.sequenceList(iSeq), optSource.sequenceRmDummies)
+
+          nbDummies = optSource.nbDummies;
+
+          fprintf('\n\nREMOVING %s DUMMIES\n\n', num2str(nbDummies));
+
+        end
+
+        % Get sequence folder path
+        sequencePath = char(fullfile(optSource.sourceDir, optSource.sequenceList(iSeq)));
+
+        % Retrieve volume files info
+        [volumesList, outputNameImage] = parseFiles('nii', sequencePath, nbDummies);
+
+        % Set output name, it takes the file name of the 1st volume of the 4D file and add subfix
+        outputNameImage = strrep(outputNameImage, '.nii', '_4D.nii');
+
+        % Retrieve sidecar json files info
+        [jsonList, outputNameJson] = parseFiles('json', sequencePath, nbDummies);
+
+        jsonFile = spm_jsonread(char(jsonList(1)));
+
+        % % % % % % LIEGE  SPECIFIC % % % % % % %
+        RT = jsonFile.acqpar.RepetitionTime / 1000;
+        % % % % % % % % % % % % % % % % % % % % %
+
+        % Set and run spm batch, input all the volumes minus the dummies if > 0
+        matlabbatch = setBatch3Dto4D(volumesList(nbDummies + 1:end, :), ...
+                                     outputNameImage, ...
+                                     optSource.dataType, ...
+                                     RT);
+
+        spm_jobman('run', matlabbatch);
+
+        if optSource.zip
+
+          % Zip and delete the and the new 4D file
+          fprintf(1, 'ZIP AND DELETE THE NEW 4D BRAIN \n\n');
+
+          gzip([sequencePath filesep outputNameImage]);
+
+          delete([sequencePath filesep outputNameImage]);
+
+        end
+
+        % Save one sidecar json file, it takes the file name of the 1st volume of the 4D file and
+        % add subfix
+        if ~isempty(jsonList)
+
+          copyfile(char(jsonList(1)), [sequencePath filesep strrep(outputNameJson, '.json', '_4D.json')]);
+
+        end
+
+        % Delete all the single volumes .nii and .json files
+        fprintf(1, 'EXTERMINATE SINGLE VOLUMES FILES \n\n');
+
+        for iDel = 1:length(volumesList)
+
+          delete(char(volumesList(iDel)));
+          delete(char(jsonList(iDel)));
+
+        end
+
+      end
+
+    end
+
+  end
+
+  newline;
+  toc;
+
+end
+
+function [fileList, outputName] = parseFiles(fileExstention, sequencePath, nbDummies)
+
+  fileList = spm_select('list', sequencePath, fileExstention);
+
+  if size(fileList, 1) > 0
+
+    outputName = fileList(nbDummies + 1, :);
+
+    fileList = strcat(sequencePath, filesep, cellstr(fileList));
+
+  else
+
+    fileList = {};
+
+    outputName = [];
+
+    newline;
+    warning('I have found 0 files with extension ''.%s'' ', fileExstention);
+    newline;
+
+  end
+
+end
+
+function initEnv()
```

Comment:
Yeah I am starting to think we need to have some sort of function that is in the root folder and sets up the path automatically, so that could be a start we can reuse later.